### DISCUSSION TITLE
[Question]: Is it a good idea to mask the Alchemy API URL when dealing with smart accounts?

### DISCUSSION THREAD

**Discussion by ocxyz** [alchemyplatform/aa-sdk#929] [Q&A] [answered]
Do I need to have a backend proxy for API URL?
```js
const client = createAlchemySmartAccountClient({
  chain: polygonAmoy,
  rpcUrl: "/api/rpc",
});
```

or it is okay to use the original URL in the frontend?
```js
const client = createAlchemySmartAccountClient({
  chain: polygonAmoy,
  rpcUrl: "https://polygon-amoy.g.alchemy.com/v2/xxxx",
});
```

Having a proxy as you know would increase our egress bill and we would like to learn if it is worth it.
Are there known security concerns we need to watch out for?

<br/>
Thanks for reading.

**Comment by moldy530** [comment, marked as answer]
That api key should be treated as a secret that you don't want to expose on the frontend, we highly recommend the proxy. otherwise, if someone gets your key then they can use it to make RPC requests on your behalf (increasing your alchemy bill)

**Reply by moldy530** [reply]
it's still worth noting that even by moving it to the backend behind a proxy, you'll probably want to add additional checks in your backend to avoid someone just calling your API directly (ie. rate limits or authentication)
